src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1481,7 +1481,12 @@ private BoundExpression BindDefaultExpression(DefaultExpressionSyntax node, Diag
                 expression = null;
                 if (node is IdentifierNameSyntax identifier)
                 {
-                    var type = BindNativeIntegerSymbolIfAny(identifier, diagnostics);
+                    // Don't bind nameof(nint) or nameof(nuint) so that ERR_NameNotInContext is reported.
+                    // The check for ArgumentSyntax is to allow the binding for something like nameof(nint.Equals).
+                    var type = IsInsideNameof && identifier.Parent is ArgumentSyntax
```

Comment:
>identifier.Parent is ArgumentSyntax [](start = 49, length = 35)

Checking for `identifier.Parent is ArgumentSyntax` seems too broad. For instance, with `nameof(typeof(nint))`, we should report "error CS8081: Expression does not have a name" for the following rather than "error CS0103: The name 'dynamic' does not exist in the current context".

Perhaps use `identifier.Parent is InvocationExpressionSyntax invocation && (invocation.Expression as IdentifierNameSyntax)?.Identifier.ContextualKind() == SyntaxKind.NameOfKeyword`.

Reply:
@cston The InvocationExpressionSyntax is the parent of the parent of the parent. So I think you meant `identifier.Parent?.Parent?.Parent`

```suggestion
                    var type = IsInsideNameof && identifier.Parent is ArgumentSyntax && identifier.Parent?.Parent?.Parent is InvocationExpressionSyntax invocation && (invocation.Expression as IdentifierNameSyntax)?.Identifier.ContextualKind() == SyntaxKind.NameOfKeyword
```